Research storage changes

Ryan180602 opened the pull request:
:cl:
tweak: Add EC variant of messenger, backpack, satchel.
tweak: Remove nt pack; make virology packs default for science.
tweak: Make a genbelt variant with science stuff for science lockers and trims some extra stuff in science lockers.
tweak: Makes science lockers start with a EC messenger. It's a EC ship after all.
/:cl:

Virology, Genetics and NT packs existed in harmony but were never properly used/were used in messy mix-matched manners.
EC Science also desperately needed a bag type that resembled more EC than CorpSci.

Corporate scientists get the Virology bags (Now Corporate Science bags, `/corpsci`) because the colour fits. And it's meant to be used off and on Torch. While the EC specific ones `/sci` are for Torch only.

MuckerMayhem commented on code/datums/outfits/jobs/_defines.dm:
```diff
@@ -16,9 +16,14 @@ backpack_overrides[/singleton/backpack_outfit/satchel]       = /obj/item/storage
+backpack_overrides[/singleton/backpack_outfit/satchel]       = /obj/item/storage/backpack/satchel/sci; \
+backpack_overrides[/singleton/backpack_outfit/messenger_bag] = /obj/item/storage/backpack/messenger/sci;
+
+#define BACKPACK_OVERRIDE_RESEARCH_CORP \
+backpack_overrides[/singleton/backpack_outfit/backpack]      = /obj/item/storage/backpack/corpsci; \
+backpack_overrides[/singleton/backpack_outfit/satchel]       = /obj/item/storage/backpack/satchel/corpsci; \
+backpack_overrides[/singleton/backpack_outfit/messenger_bag] = /obj/item/storage/backpack/messenger/corpsci;
```
Is there a point to having this and the virology backpack override if they're both the same now?

Ryan180602 replied:
`_CORP` is meant to be spawnable by contractor scientists, whereas the non-corp is for EC only.

SierraKomodo replied:
`BACKPACK_OVERRIDE_VIROLOGY` is redundant with `BACKPACK_OVERRIDE_RESEARCH_CORP` is what Mucker means.

Ryan180602 replied:
removed `_VIROLOGY`, am stupid.

MuckerMayhem commented on code/game/objects/items/weapons/storage/belt.dm:
```diff
@@ -318,6 +318,16 @@
+	new /obj/item/folder(src)
+	new /obj/item/device/taperecorder(src)
+	new /obj/item/device/camera(src)
+	new /obj/item/taperoll/research(src)
+	new /obj/item/device/tape/random(src)
+	new /obj/item/device/soulstone(src)
+
```
```suggestion
	new /obj/item/taperoll/research(src)
	new /obj/item/device/tape/random(src)

```
I don't think you meant to have that in there.

Ryan180602 replied:
free soulstone for your internship at the SEV Torch

(no it wasn't, I copied it from the soulstone belt)

SierraKomodo commented on code/game/objects/items/weapons/storage/belt.dm:
```diff
@@ -318,6 +318,15 @@
 		/obj/item/device/tape
 		)
 
+/obj/item/storage/belt/general/full/New()
```
Use `Initialize()` unless you absolutely _must_ use `New()`.

Ryan180602 replied:
fixe

SierraKomodo reviewed (dismissed):
Test failure:

```
## UNIT_TEST ##: [SUBSYSTEM - ATOMS: Shall have no bad init calls]: Path : /obj/item/storage/belt/general/full 
- Didn't return an Initialize hint
```

Ryan180602 committed:
research storage changes

add EC variant of messenger, backpack, satchel; remove nt pack; make virology packs default for science; name viropacks into corpscipacks; make a genbelt variant with science stuff for science lockers
a


a

Ryan180602 commented:
Fixed.
